### PR TITLE
netevent: init at 20201018

### PR DESCRIPTION
**Pull request by rgrunbla**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 101019` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netevent</li>
  </ul>
</details>

**Review by RaitoBezarius — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 101019` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netevent</li>
  </ul>
</details>

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [inline pkgs/tools/inputmethods/netevent/default.nix:21]: Why override the configurePhase - it already runs ./configure. At the very least you should set --prefix=$out and remove the overridden installPhase below

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [inline pkgs/tools/inputmethods/netevent/default.nix:18]: It looks like this isn't necessary, netevent already uses rst2man from PATH under this name

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [inline pkgs/tools/inputmethods/netevent/default.nix:3]: No need for rec

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [inline pkgs/tools/inputmethods/netevent/default.nix:10]: It looks like there are some release tags upstream; Debian has some numbered releases as well; can you use one of those instead of an arbitrary hash? Alternatively maybe you want to file an issue upstream asking for a release

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [inline pkgs/tools/inputmethods/netevent/default.nix:33]: This all seems unnecessary, seems to be done by the package itself if you set --prefix appropriately in configure.

**Review by catern — CHANGES_REQUESTED** [review]
The commit message/PR title should be netevent: init at 20201018

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:21]: The ./configure is not supporting classical configure options, which means not overriding the configurePhase results in errors.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:18]: When not setting RST2MAN to this value, for some reason, the configure (which is not a classical configure) does not find the binary. This ensure the check is skipped and rst2man is effectively used.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:10]: The latest release dates from 10 years ago, which is why I'm not using it. I can ask for a release (but the goal is primarily packaging the tool), but in the mean time, i'm using the latest master branch revision.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:33]: When setting the --prefix in the configure (which is not a classical configure), the data does not end up in the good directory, so I choose to do it manually.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix]: Fixed.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (catern) [reply pkgs/tools/inputmethods/netevent/default.nix:33]: From looking at the configure script, the directories seem fine if you just set prefix=$out. What's wrong concretely?

Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:33]: Here what I get:

```
 nix-build -A netevent
these derivations will be built:
  /nix/store/bxh7y6n33n4n46cyjn8fldbvl2idif1v-netevent-20201018.drv
building '/nix/store/bxh7y6n33n4n46cyjn8fldbvl2idif1v-netevent-20201018.drv'...
unpacking sources
unpacking source archive /nix/store/1k2ming61x2ga1kz43x39c5lg4q0ph1a-source
source root is source
patching sources
configuring
Checking for rst2man...rst2man
Checking for UI_DEV_SETUP (kernel >= 4.4)...ok
building
build flags: SHELL=/nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/main.o src/main.cpp -MMD -MT src/main.o -MF src/main.d
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/daemon.o src/daemon.cpp -MMD -MT src/daemon.o -MF src/daemon.d
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/writer.o src/writer.cpp -MMD -MT src/writer.o -MF src/writer.d
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/reader.o src/reader.cpp -MMD -MT src/reader.o -MF src/reader.d
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/socket.o src/socket.cpp -MMD -MT src/socket.o -MF src/socket.d
g++ -g -Wall -Werror -Wno-unknown-pragmas  -std=c++14 -c -o src/bitfield.o src/bitfield.cpp -MMD -MT src/bitfield.o -MF src/bitfield.d
g++ -g  -o netevent src/main.o src/daemon.o src/writer.o src/reader.o src/socket.o src/bitfield.o
rst2man doc/netevent.rst doc/netevent.1
installing
install flags: SHELL=/nix/store/k8p54jg8ipvnfz435mayf5bnqhw4qqap-bash-4.4-p23/bin/bash install
make install-do
make[1]: Entering directory '/build/source'
install -dm755
install: missing file operand
Try 'install --help' for more information.
make[1]: *** [Makefile:67: install-do] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:64: install] Error 2
builder for '/nix/store/bxh7y6n33n4n46cyjn8fldbvl2idif1v-netevent-20201018.drv' failed with exit code 2
error: build of '/nix/store/bxh7y6n33n4n46cyjn8fldbvl2idif1v-netevent-20201018.drv' failed
```

Given the state of the configure file, I prefer to move things manually around.

Comment (rgrunbla) [reply pkgs/tools/inputmethods/netevent/default.nix:33]: Any comment @catern ? I'm not really sure how I should proceed now, as it's my first package request i there, and because the points you made are good, but at the same time I'm not sure they can be addressed by the package management. Do you think I should resolve the conversation? Do I have to wait for more reviews?

Comment (catern) [reply pkgs/tools/inputmethods/netevent/default.nix:33]: Since you're working off the master branch of the package anyway, maybe you can try and fix the configure script with a patch upstream.

File: maintainers/maintainer-list.nix
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:7190]: Please put this into a separate commit with the message ``maintainers: add rgrunbla``.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (SuperSandro2000) [reply pkgs/tools/inputmethods/netevent/default.nix:18]: Please export this in configure.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/netevent/default.nix:26]: Please use install here.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/netevent/default.nix:29]: Please use installManPage.

File: pkgs/tools/inputmethods/netevent/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/netevent/default.nix:40]:
```suggestion
    maintainers = with maintainers; [ rgrunbla ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101019` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netevent</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101019` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>netevent</li>
  </ul>
</details>

**Pull request merged**
